providers/efa/verbs.c

```diff
@@ -88,6 +91,30 @@ int efa_query_device_ex(struct ibv_context *context,
 	return 0;
 }
 
+int efadv_query_device(struct ibv_context *ibvctx,
+		       struct efadv_device_attr *attr)
+{
+	struct efa_context *ctx = to_efa_context(ibvctx);
+	struct efa_dev *dev = to_efa_dev(ibvctx->device);
+	uint64_t comp_mask_out = 0;
+
+	if (!is_efa_dev(ibvctx->device))
+		return EOPNOTSUPP;
+
+	if (!field_avail(attr, inline_buf_size, attr->inlen))
```

Comment:
This needs to zero the output structure - trailing data and reserved bytes that are not written to need to be zero'd

Reply:
Done